ReleaseProcessScript.New.Tests/CommandLineGitClientTests.cs

```diff
@@ -0,0 +1,310 @@
+
+    var client = new CommandLineGitClient(_dir);
+
+    var ancestor = client.GetAncestors("develop", "release");
+
+    Assert.That(ancestor, Does.Contain("develop"));
+    Assert.That(ancestor, Does.Contain("release/v1.0.1"));
+    Assert.That(ancestor, Does.Not.Contain("master"));
+  }
+
+  [Test]
+  public void GetAncestors_ReleaseOnSameCommitAsDevelop_ReturnsTwo ()
+  {
+    TestAddCommit();
+    ExecuteGitCommand("checkout -b develop -q");
+    ExecuteGitCommand("checkout -b developer");
+    ExecuteGitCommand("checkout -b release/v1.0.1 -q");
+    var client = new CommandLineGitClient(_dir);
+
+    var ancestor = client.GetAncestors("develop");
+
+    Assert.That(ancestor, Does.Contain("develop"));
+    Assert.That(ancestor, Does.Contain("developer"));
+    Assert.That(ancestor, Does.Not.Contain("release/v1.0.1"));
+  }
+
+  [Test]
+  public void GetAncestor_SeveralBranchesFromMaster_ReturnsOnlyOne ()
+  {
+    TestAddCommit();
+    ExecuteGitCommand("checkout -b develop -q");
+    TestAddCommit();
+    ExecuteGitCommand("checkout master");
+    ExecuteGitCommand("checkout -b developer -q");
+    TestAddCommit();
+    ExecuteGitCommand("checkout -b release/v1.0.1 -q");
+    ExecuteGitCommand("checkout -b developeria -q");
+    TestAddCommit();
+    ExecuteGitCommand("checkout -b release/v1.0.2 -q");
+    var client = new CommandLineGitClient(_dir);
+
+    var ancestor = client.GetAncestors("develop");
+
+    Assert.That(ancestor, Does.Contain("developer"));
+    Assert.That(ancestor, Does.Contain("developeria"));
+  }
+
+  private static void ExecuteGitCommand (string argument)
+  {
+    using var command = Process.Start("git", argument);
+    command.WaitForExit();
+  }
+
+  private static string ExecuteGitCommandWithOutput (string argument)
+  {
+    var psi = new ProcessStartInfo("git", argument);
+    psi.UseShellExecute = false;
+    psi.RedirectStandardOutput = true;
+    psi.RedirectStandardError = true;
+
+    using var command = Process.Start(psi);
+    command.WaitForExit();
+
+    var output = command.StandardOutput.ReadToEnd();
+
+    return output;
+  }
+
+  private static void DeleteDirectory (string target_dir)
+  {
+    var files = Directory.GetFiles(target_dir);
+    var dirs = Directory.GetDirectories(target_dir);
+
+    foreach (var file in files)
+    {
+      File.SetAttributes(file, FileAttributes.Normal);
+      File.Delete(file);
+    }
+
+    foreach (var dir in dirs)
+      DeleteDirectory(dir);
+
+    Directory.Delete(target_dir, false);
+  }
+
+  private static string AddRandomFile (string dir)
+  {
+    var random = new Random().Next();
+    var path = dir + "\\" + random;
+
+    using var file = File.Create(path);
+    var info = new UTF8Encoding(true).GetBytes(random.ToString());
+    file.Write(info, 0, info.Length);
+
+    File.SetAttributes(path, FileAttributes.Normal);
+
+    return random.ToString();
+  }
+
+  private void TestAddCommit (string s = "")
+  {
+    var fileName = AddRandomFile(_dir);
```

Comment:
Not sure if this is needed here. The `--allow-empty` should be enough I thought?